### PR TITLE
Feature/menu

### PR DESCRIPTION
**Pull request by BWalusiak**
add menu and 2am lobby

### CODE REVIEW

File: client/core/src/com/cyberbot/bomberman/managers/GameScreenManager.java [hunk 1]
```diff
@@ -1,51 +0,0 @@
...
-        this.app = app;
-        initScreens();
-
-        setScreen(ScreenState.GAME);
-    }
-
-    private void initScreens() {
-        screens = new HashMap<>();
-        try {
-            // TODO: Load game screen when a game start packet is received with proper data
```
Comment (RouNNdeL) [inline client/core/src/com/cyberbot/bomberman/managers/GameScreenManager.java:38]: Add this TODO back

File: client/core/src/com/cyberbot/bomberman/screens/AbstractScreen.java [hunk 2]
```diff
@@ -4,13 +4,16 @@
...
+import com.cyberbot.bomberman.controllers.GameScreenController;
 
 public abstract class AbstractScreen implements Screen {
 
     protected final Client app;
+    protected final GameScreenController gameScreenController;
 
-    public AbstractScreen(final Client app) {
+    public AbstractScreen(final Client app, GameScreenController gameScreenController) {
         this.app = app;
```
Comment (RouNNdeL) [inline client/core/src/com/cyberbot/bomberman/screens/AbstractScreen.java]: `app` was deemed useless, any reasons to keep it?

**Review by RouNNdeL — CHANGES_REQUESTED** [review]

File: client/core/src/com/cyberbot/bomberman/controllers/ScreenChangeInterface.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.cyberbot.bomberman.controllers;
+
+import com.cyberbot.bomberman.screens.ScreenState;
+
+public interface ScreenChangeInterface {
+    void setScreen(ScreenState state);
+    public void setScreen(ScreenState state, String playerName, boolean isOwner);
```
Comment (RouNNdeL) [inline client/core/src/com/cyberbot/bomberman/controllers/ScreenChangeInterface.java]: Implement separate methods for each `ScreenState`, this allows for better metadata handling. Ex. 
```java
setMenuScreen()
setLobbyScreen(playerName: String, isOwner: boolean)
```

File: client/core/src/com/cyberbot/bomberman/controllers/GameScreenController.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            default:
+                throw new IllegalStateException("Unexpected value: " + state);
+        }
+    }
+
+    @Override
+    public void setScreen(ScreenState state, String playerName, boolean isOwner) {
+        if (!state.equals(ScreenState.LOBBY)){
+            throw new IllegalArgumentException("Only lobby screen can contain player nickname");
+        }
```
Comment (RouNNdeL) [inline client/core/src/com/cyberbot/bomberman/controllers/GameScreenController.java]: Well...
> Implement separate methods for each ScreenState, this allows for better metadata handling.

File: client/core/src/com/cyberbot/bomberman/controllers/GameScreenController.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+import com.cyberbot.bomberman.core.models.tiles.MissingLayersException;
+import com.cyberbot.bomberman.screens.*;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.HashMap;
+
+public final class GameScreenController implements ScreenChangeInterface{
```
Comment (RouNNdeL) [inline client/core/src/com/cyberbot/bomberman/controllers/GameScreenController.java]: Ctrl+Alt+L

**Review by RouNNdeL — CHANGES_REQUESTED** [review]

File: client/core/src/com/cyberbot/bomberman/controllers/GameScreenController.java (hunk 5 above)
Comment (BWalusiak) [reply client/core/src/com/cyberbot/bomberman/controllers/GameScreenController.java]: fixed

File: client/core/src/com/cyberbot/bomberman/controllers/GameScreenController.java (hunk 4 above)
Comment (BWalusiak) [reply client/core/src/com/cyberbot/bomberman/controllers/GameScreenController.java]: fixed

File: client/core/src/com/cyberbot/bomberman/controllers/ScreenChangeInterface.java (hunk 3 above)
Comment (BWalusiak) [reply client/core/src/com/cyberbot/bomberman/controllers/ScreenChangeInterface.java]: fixed

File: client/core/src/com/cyberbot/bomberman/screens/AbstractScreen.java (hunk 2 above)
Comment (BWalusiak) [reply client/core/src/com/cyberbot/bomberman/screens/AbstractScreen.java]: fixed

File: client/core/src/com/cyberbot/bomberman/managers/GameScreenManager.java (hunk 1 above)
Comment (BWalusiak) [reply client/core/src/com/cyberbot/bomberman/managers/GameScreenManager.java:38]: fixed

**Review by RouNNdeL — APPROVED** [review]

**Pull request merged**